# 03 — KPIs

Indicadores clave para la capa ejecutiva del dashboard.

In [1]:
import pandas as pd

df = pd.read_csv('../datasets/firewall_clean.csv')
logins = pd.read_csv('../datasets/login_attempts.csv')

In [2]:
kpis = {
    'Eventos totales': len(df),
    'IPs únicas': df['src_ip'].nunique(),
    'Intentos fallidos (login)': int((logins['success'].astype(str).str.lower()=='false').sum()),
    'Eventos críticos': int((df['severity']=='critica').sum()),
    'Eventos denegados': int((df['action']=='deny').sum()),
    'Top atacante': df['src_ip'].value_counts().idxmax(),
    'Eventos del top atacante': int(df['src_ip'].value_counts().max()),
}
for k, v in kpis.items():
    print(f'{k:<28}: {v}')

Eventos totales             : 18031
IPs únicas                  : 124
Intentos fallidos (login)   : 5085
Eventos críticos            : 5445
Eventos denegados           : 8700
Top atacante                : 45.21.111.60
Eventos del top atacante    : 1281


## Tabla de KPIs

| KPI | Valor |
|-----|------|
| Eventos totales | 18,031 |
| IPs únicas | 124 |
| Intentos fallidos (login) | 5,085 |
| Eventos críticos | 5,445 |
| Top atacante | 45.21.111.60 (1,281 eventos) |